Services/TMS/Mailing/classes/class.ilTMSMailContextCourseCreation.php

```diff
@@ -0,0 +1,170 @@
+<?php
+declare(strict_types=1);
+
+use \ILIAS\TMS\Mailing;
+use \ILIAS\TMS\CourseCreation\Request;
+
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> */
+
+/**
+ * Course-related placeholder-values
+ */
+class ilTMSMailContextCourseCreation implements Mailing\MailContext {
+	private static $PLACEHOLDER = array(
+		'COURSE_TITLE' => 'placeholder_desc_crs_title',
+		'COURSE_START_DATE' => 'placeholder_desc_crs_startdate',
+		'COURSE_END_DATE' => 'placeholder_desc_crs_enddate'
+	);
+
+	/**
+	 * @var int
+	 */
+	protected $request;
+
+	/**
+	 * @var ilLanguage
+	 */
+	protected $g_lang;
+
+	public function __construct(Request $request, \ilLanguage $lng)
+	{
+		$this->request = $request;
+		$this->lng = $lng;
+		$this->lng->loadLanguageModule("tms");
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function valueFor($placeholder_id, $contexts = array())
```

Comment:
return type could be string here if the changes downside take place and adjusting the mailcontext. also by adjusting mailcontext we can typehint in php7 style.